cmd/otel-agent/subcommands/run/command.go

```diff
@@ -96,10 +96,16 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 	if err != nil && err != agentConfig.ErrNoDDExporter {
 		return err
 	}
+	uris := append(params.ConfPaths, params.Sets...)
 	if err == agentConfig.ErrNoDDExporter {
 		return fxutil.Run(
-			fx.Provide(func() []string {
-				return append(params.ConfPaths, params.Sets...)
+			fx.Supply(uris),
+			fx.Provide(func() optional.Option[coreconfig.Component] {
+				return optional.NewNoneOption[coreconfig.Component]()
+			}),
```

Comment:
You could replace this with the shorter version:
```suggestion
			fx.Supply(optional.NewNone[coreconfig.Component]),
```